src/state/zustand/swapperStore/amountSelectors.ts

```diff
@@ -0,0 +1,394 @@
+        buyAssetPrecision,
+      ),
+      sellAssetPrecision,
+    )
+  },
+)
+
+export const selectTotalTradeFeeSellAssetBaseUnit = createSelector(
+  selectSellAssetTradeFeeSellAssetBaseUnit,
+  selectBuyAssetTradeFeeSellAssetBaseUnit,
+  (sellAssetTradeFeeSellAssetBaseUnit, buyAssetTradeFeeSellAssetBaseUnit): string | undefined => {
+    if (!sellAssetTradeFeeSellAssetBaseUnit || !buyAssetTradeFeeSellAssetBaseUnit) return undefined
+    return bnOrZero(sellAssetTradeFeeSellAssetBaseUnit)
+      .plus(buyAssetTradeFeeSellAssetBaseUnit)
+      .toString()
+  },
+)
+
+export const selectTotalTradeFeeBuyAssetBaseUnit = createSelector(
+  selectSellAssetTradeFeeBuyAssetBaseUnit,
+  selectBuyAssetTradeFeeBuyAssetBaseUnit,
+  (sellAssetTradeFeeBuyAssetBaseUnit, buyAssetTradeFeeBuyAssetBaseUnit): string | undefined => {
+    if (!sellAssetTradeFeeBuyAssetBaseUnit || !buyAssetTradeFeeBuyAssetBaseUnit) return undefined
+    return bnOrZero(sellAssetTradeFeeBuyAssetBaseUnit)
+      .plus(buyAssetTradeFeeBuyAssetBaseUnit)
+      .toString()
+  },
+)
+
+export const selectTotalTradeFeeBuyAsset = createSelector(
+  selectTotalTradeFeeBuyAssetBaseUnit,
+  (state: SwapperState) => state.buyAsset?.precision,
+  (totalTradeFeeBuyAssetBaseUnit, buyAssetPrecision): string | undefined => {
+    if (!totalTradeFeeBuyAssetBaseUnit || !buyAssetPrecision) return undefined
+    return fromBaseUnit(totalTradeFeeBuyAssetBaseUnit, buyAssetPrecision)
+  },
+)
+
+export const selectSellAmountPlusFeesBaseUnit = createSelector(
+  selectSellAmountBeforeFeesBaseUnit,
+  selectTotalTradeFeeSellAssetBaseUnit,
+  (sellAmountBeforeFeesBaseUnit, totalTradeFeeSellAssetBaseUnit): string | undefined => {
+    if (!sellAmountBeforeFeesBaseUnit || !totalTradeFeeSellAssetBaseUnit) return undefined
+    return bnOrZero(sellAmountBeforeFeesBaseUnit).plus(totalTradeFeeSellAssetBaseUnit).toString()
+  },
+)
+
+export const selectBuyAmountBeforeFeesFiat = createSelector(
+  selectBuyAmountBeforeFeesBaseUnit,
+  (state: SwapperState) => state.buyAsset?.precision,
+  (state: SwapperState) => state.buyAssetFiatRate,
+  (buyAmountBeforeFeesBaseUnit, buyAssetPrecision, buyAssetFiatRate): string | undefined => {
+    if (!buyAssetPrecision || !buyAssetFiatRate) return undefined
+    return bnOrZero(fromBaseUnit(buyAmountBeforeFeesBaseUnit, buyAssetPrecision))
+      .times(buyAssetFiatRate)
+      .toFixed(2)
+  },
+)
+
+export const selectSellAmountBeforeFeesFiat = createSelector(
+  selectSellAmountBeforeFeesBaseUnit,
+  (state: SwapperState) => state.sellAsset?.precision,
+  (state: SwapperState) => state.sellAssetFiatRate,
+  (sellAmountBeforeFeesBaseUnit, sellAssetPrecision, sellAssetFiatRate): string | undefined => {
+    if (!sellAssetPrecision || !sellAssetFiatRate) return undefined
+    return bnOrZero(fromBaseUnit(sellAmountBeforeFeesBaseUnit, sellAssetPrecision))
+      .times(sellAssetFiatRate)
+      .toFixed(2)
```

Comment:
We might want to let consumers handle the `toFixed()`, to keep selectors as atomic as possible WRT data they return

Reply:
Yes, this is definitely the right way to handle this. Updated all selectors to use full precision in https://github.com/shapeshift/web/pull/4138/commits/10b9d8bcefe0a0c36807e69f55d674898f971837, leaving the 2-decimal rounding to the UI layer.